velox/expression/ComplexWriterTypes.h

```diff
@@ -773,10 +773,7 @@ class MapWriter {
     keysWriter_->ensureSize(1);
     valuesWriter_->ensureSize(1);
 
-    VELOX_DCHECK(
-        keysVector_->size() == valuesVector_->size(),
-        "expect map keys and value vector sized to be synchronized");
-    capacity_ = keysVector_->size();
+    capacity_ = std::min(keysVector_->size(), valuesVector_->size());
```

Comment:
the code in this file seems to have an implicit assumption that capacity_ represents the capacity of both values and keys vector. It is then co-related with other state variables like innerOffset_ that represent the offset within **both** values and keys vector.
therefore, if now
`innerOffset_` is **max**(keys.size(), values.size()) and
`capacity_` is **min**(keys.size(), values.size())

then code like this:
```
  void reserve(vector_size_t size) {
    auto currentSize = size + innerOffset_;

    if (UNLIKELY(currentSize > capacity_)) {
      capacity_ = std::pow(2, std::ceil(std::log2(currentSize)));

      keysWriter_->ensureSize(capacity_);
      valuesWriter_->ensureSize(capacity_);
    }
  }
```
and comments like this
```
// The offset within the children vectors at which this map starts.
  vector_size_t innerOffset_ = 0;
```

will sound contradictory because here innerOffset_ > capacity_  and innerOffset_ should not point to anything beyond capacity.

Since there is a 1-1 mapping between key-value, and anything beyond **min**(keys.size(), values.size()) is irrelevant, we can set  

`innerOffset_` to **max**(keys.size(), values.size()) as well